### ISSUE TITLE
In Docker, clicking into to camera in home app temporarily kills Homebridge service

### ISSUE THREAD

**Issue by gadgetfreak98** [KhaosT/homebridge-nest-cam#63]
Hi. Love, love, love what you are enabling here!  I had this working fine under a regular terminal setup, but for various reasons, I ended up needing to resort to using Docker -- which was way easier in many ways.  

The issue, however, is that clicking through the preview window of my camera in the Home app does two things:

1. It throws up the "camera unavailable" error and, oddly:
2. It temporarily kills the Homebridge service according to the Docker Homebridge Server Status page.  When I exit back out of the detailed camera view to the regular list of Home items, the server reconnects and all is well.  The camera in the summary Home app tab also then reconnects.

So basically I can't click through to watch in full screen on my phone without killing the Homebridge service temporarily.  I googled around but couldn't find anything talking about this?  Thoughts?  So close right now!

Thanks!

**Comment by KhaosT** [comment]
Make sure your Docker has ffmpeg in there.

**Comment by gadgetfreak98** [comment]
I tried to install it, but I'm not sure I did it correctly.  Any recommendation about how to ensure ffmpeg is properly installed in Docker?

**Comment by KhaosT** [comment]
Err… you can attach a console to Docker and run ffmpeg to see if that actually runs.

**Comment by gadgetfreak98** [comment]
Not quite sure how to do that... am googling around

**Comment by gadgetfreak98** [comment]
Just checking back on this.  Would you mind suggesting a place to get further directions on this?  Or provide them if easy to do so?  thanks.

**Comment by KhaosT** [comment]
Something like this.

https://github.com/oznu/docker-homebridge/issues/65

**Comment by gadgetfreak98** [comment]
Thanks. I'm still at a bit of a loss as to how to actually modify the startup.sh file in Docker on a Raspberry Pi.

**Comment by dp917** [comment]
@gadgetfreak98 
You have to enable ffmpeg for docker by adding the environment variable PACKAGES=ffmpeg. Can be easily done if you're using portainer to manage docker. Had same issues as you and this fixed it. I also made sure my machine had ffmpeg installed and with using terminal in homebridge with configuix just to cover everything.

**Comment by gadgetfreak98** [comment]
> Something like this. [oznu/docker-homebridge#65](https://github.com/oznu/docker-homebridge/issues/65)
> […](#)
> Sent from my iPhone
> On Feb 10, 2019, at 3:58 PM, gadgetfreak98 ***@***.***> wrote: Just checking back on this. Would you mind suggesting a place to get further directions on this? Or provide them if easy to do so? thanks. 


I finally figured out how to do this after some time away.... Just couldn't initially figure out what I was supposed to do in the directions in that link.  But after having start again from scratch using Docker this time on a Synology vs. a Pi, I understood where the ffmpeg script was to go, and it worked! Thanks!
